lib/scholar/metrics/classification.ex

```diff
@@ -677,7 +739,122 @@ defmodule Scholar.Metrics.Classification do
     end
   end
 
-  defnp precision_recall_fscore_n(y_true, y_pred, beta, num_classes, average) do
+  @doc """
+  Calculates precision, recall, F-score and support for each
+  class. It also supports a `beta` argument which weights
+  recall more than precision by it's value.
+
+  ## Options
+
+  #{NimbleOptions.docs(@precision_recall_fscore_support_schema)}
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([0, 1, 1, 1, 1, 0, 2, 1, 0, 1], type: :u32)
+      iex> y_pred = Nx.tensor([0, 2, 1, 1, 2, 2, 2, 0, 0, 1], type: :u32)
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3)
+      {#Nx.Tensor<
+       f32[3]
+       [0.6666666865348816, 1.0, 0.25]
+      >,
+      #Nx.Tensor<
+       f32[3]
+       [0.6666666865348816, 0.5, 1.0]
+      >,
+      #Nx.Tensor<
+       f32[3]
+       [0.6666666865348816, 0.6666666865348816, 0.4000000059604645]
+      >,
+      #Nx.Tensor<
+       u64[3]
+       [3, 6, 1]
+      >}
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :macro)
+      {#Nx.Tensor<
+       f32[3]
+       [0.6666666865348816, 1.0, 0.25]
+      >,
+      #Nx.Tensor<
+       f32[3]
+       [0.6666666865348816, 0.5, 1.0]
+      >,
+      #Nx.Tensor<
+       f32
+       0.5777778029441833
+      >,
+      #Nx.Tensor<
+       f32
+       NaN
+      >}
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :weighted)
+      {#Nx.Tensor<
+       f32[3]
+       [0.6666666865348816, 1.0, 0.25]
+      >,
+      #Nx.Tensor<
+       f32[3]
+       [0.6666666865348816, 0.5, 1.0]
+      >,
+      #Nx.Tensor<
+       f32
+       0.6399999856948853
+      >,
+      #Nx.Tensor<
+       f32
+       NaN
+      >}
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :micro)
+      {#Nx.Tensor<
+       f32
+       0.6000000238418579
+      >,
+      #Nx.Tensor<
+       f32
+       0.6000000238418579
+      >,
+      #Nx.Tensor<
+       f32
+       0.6000000238418579
+      >,
+      #Nx.Tensor<
+       f32
+       NaN
+      >}
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(Nx.tensor([1, 0, 1, 0]), Nx.tensor([0, 1, 0, 1]), beta: 2, num_classes: 2, average: :none)
```

Comment:
Please add new lines before `iex` whenever you can, so ExDoc formats them as distinct examples. Also, if you can break this line into several ones (for exmaple, by assigning the tensors and opts to variables, it can help with readability).